src/VisualStudio/Core/Def/Implementation/NavigateTo/VisualStudioNavigateToItemProviderFactory.cs

```diff
@@ -25,8 +26,15 @@ public VisualStudioNavigateToItemProviderFactory(VisualStudioWorkspace workspace
             _workspace = workspace;
         }
 
-        public bool TryCreateNavigateToItemProvider(IServiceProvider serviceProvider, out INavigateToItemProvider provider)
+        public bool TryCreateNavigateToItemProvider(IServiceProvider serviceProvider, out INavigateToItemProvider? provider)
         {
+            // Let LSP handle goto when running under the LSP editor.
+            if (_workspace.Services.GetRequiredService<IWorkspaceContextService>().IsInLspEditorContext())
```

Comment:
i'm confused by this.  This is navigate-to, not go-to.  They are different features (the latter being inferior in a lot of ways).  I would not want to disable navigate-to when using LSP, unless LSP can actually power it.

Reply:
discussed a bit offline - when the feature flag is on, LSP can power both navigateto and vs search